### PR TITLE
Partially fix Url parsing in CDN job

### PR DESCRIPTION
**Pull request by ryuyu**
Parsing package id and version out of the urls in the cdn logs is currently dependent on the requested file path (nupkg). This introduces ambiguity in the cases where the package id ends with a .{number} pattern.
For example, a package with id Pack.321 of version 1.0.0 has nupkg Pack.321.1.0.0.nupkg. Looking at just the nupkg name, it is ambiguous whether the package id is Pack v321.1.0.0 or Pack.321 v1.0.0.

This change partially fixes this by checking earlier parts of the url for a package id and version to verify against (ie a v3 url).

This is also likely related to https://github.com/NuGet/NuGetGallery/issues/3378

@xavierdecoster @skofman1

### CODE REVIEW

File: src/Stats.AzureCdnLogs.Common/PackageDefinition.cs [hunk 1]
```diff
@@ -27,9 +27,24 @@ public static PackageDefinition FromRequestUrl(string requestUrl)
 
 
             var urlSegments = requestUrl.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+            var maybePackageId = urlSegments[urlSegments.Length - 3];
```
Comment (skofman1) [inline src/Stats.AzureCdnLogs.Common/PackageDefinition.cs]: we shouldn't make assumptions regarding path length. urlSegments[urlSegments.Length - 3] might not exist.

File: src/Stats.AzureCdnLogs.Common/PackageDefinition.cs [hunk 2]
```diff
@@ -27,9 +27,24 @@ public static PackageDefinition FromRequestUrl(string requestUrl)
...
+            var maybePackageVersion = urlSegments[urlSegments.Length - 2];
+            var reconstructV3FileName = maybePackageId + "." + maybePackageVersion + _nupkgExtension;
             var fileName = urlSegments.Last();
 
-            if (fileName.EndsWith(_nupkgExtension))
+            var isV3Url = String.Compare(fileName, reconstructV3FileName, ignoreCase: true) == 0;
```
Comment (skofman1) [inline src/Stats.AzureCdnLogs.Common/PackageDefinition.cs]: will this work for different cultures?

File: src/Stats.AzureCdnLogs.Common/PackageDefinition.cs [hunk 3]
```diff
@@ -27,9 +27,24 @@ public static PackageDefinition FromRequestUrl(string requestUrl)
...
+
+            if (isV3Url)
```
Comment (skofman1) [inline src/Stats.AzureCdnLogs.Common/PackageDefinition.cs]: We can move the parsing logic to a different method, so the code will stay simple

File: src/Stats.AzureCdnLogs.Common/PackageDefinition.cs (hunk 2 above)
Comment (ryuyu) [inline src/Stats.AzureCdnLogs.Common/PackageDefinition.cs]: Looks like we use OrdinalIgnoreCase comparison elsewhere. Fixed.

File: src/Stats.AzureCdnLogs.Common/PackageDefinition.cs (hunk 3 above)
Comment (ryuyu) [inline src/Stats.AzureCdnLogs.Common/PackageDefinition.cs]: Moved.

File: src/Stats.AzureCdnLogs.Common/PackageDefinition.cs (hunk 1 above)
Comment (ryuyu) [inline src/Stats.AzureCdnLogs.Common/PackageDefinition.cs]: Added a check for url segments being too short.

**Comment by skofman1** [comment]
Lets wait for @xavierdecoster to review before merging.

**Comment by shishirx34** [comment]
This hasn't been merged yet. Any more related work pending here?

**Comment by xavierdecoster** [comment]
@ryuyu I think a rebase on `dev` is needed here to clean up the diff/PR

**Comment by joelverhagen** [comment]
Closing since we have moved to the new statistics pipeline.

**Pull request closed without merging**